code/modules/mob/living/living.dm

```diff
@@ -443,6 +443,12 @@
 	if (s_active && !(s_active.ClickAccessible(src, depth=STORAGE_VIEW_DEPTH) || s_active.Adjacent(src)))
 		s_active.close(src)
 
+	for(var/mob/M in oview(src))
```

Comment:
Replace this with viewers() and then you will only get mobs and can remove the type on the for and do a istype for.